hw09-byteCodes/src/main/java/ru/otus/IoC.java

```diff
@@ -0,0 +1,56 @@
+package ru.otus;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.List;
+
+class IoC {
+    private static final List<Method> methodList = new ArrayList<>();
+    static AnyClassInterface createMyClass() {
+        InvocationHandler handler = new DemoInvocationHandler(new AnyClass());
+        getMethods();
```

Comment:
Лучше без побочных эффектов:
```suggestion
        methodList = getMethods(AnyClassInterface.class);
```

или переименовать:
```suggestion
        cacheMethods();
```

`get` - подразумевает, что вы что-то возвращаете из метода